ClientApp/src/app/services/login.service.ts

```diff
@@ -1,24 +1,45 @@
 import { Injectable } from '@angular/core';
 
-import { BaseApiService, ApiOptions } from '../services/api/base-api.service';
+import { UserProviderService } from '../services/api/user-provider.service';
+import { UserService } from '../services/user.service';
+import { SessionService } from '../services/session.service';
+import { User } from '../models/user.model';
 
 @Injectable({
   providedIn: 'root',
 })
-export class LoginService extends BaseApiService {
-  constructor(protected apiOptions: ApiOptions) {
-    super(apiOptions);
-  }
+export class LoginService {
+  constructor(
+    private userProvider: UserProviderService,
+    private userService: UserService,
+    private sessionService: SessionService,
+  ) {}
 
-  async authencicateUser(user: any) {
-    return await super.post('authenticate-user', user).toPromise();
+  async authencicateUser(email: string, password: string): Promise<boolean> {
+    return this.userProvider
+      .login(email, password)
+      .then(async (response: any) => {
+        if (typeof response === 'string') {
+          this.sessionService.setToken(response);
+          await this.userService.fetchUser(response);
+          return true;
+        }
+        return false;
+      });
   }
 
   async registerUser(user: any) {
-    return await super.post('register-user', user).toPromise();
+    return this.userProvider.register(user).then(async (response: string) => {
+      if (typeof response === 'string') {
+        this.authencicateUser(user.Email, user.Password);
+        return true;
+      }
+      return false;
+    });
```

Comment:
You need to Log In after signing up.
=> You don't need the authenticate function in the sign up. 😤